easybuild/tools/modules.py

```diff
@@ -718,6 +720,86 @@ def loaded_modules(self):
 
         return loaded_modules
 
+    def check_loaded_modules(self):
+        """
+        Check whether any (EasyBuild-generated) modules are loaded already in the current session
+        """
+        ignored_keys = [get_software_root_env_var_name(x) for x in build_option('allow_loaded_modules') or [] if x]
+
+        eb_module_keys = []
+        for key in os.environ:
+            if key.startswith(ROOT_ENV_VAR_NAME_PREFIX) and key not in ignored_keys:
+                eb_module_keys.append(key)
+
+        if eb_module_keys:
+            loaded_modules = self.loaded_modules()
+
+            # try to track down modules that define the $EBROOT* environment variables that were found
+            loaded_eb_modules = []
+            for loaded_module in loaded_modules:
+                out = self.show(loaded_module)
+                for key in eb_module_keys[:]:
+                    if key in out:
+                        loaded_eb_modules.append(loaded_module)
+                        eb_module_keys.remove(key)
+
+            # warn about $EBROOT* environment variables without matching loaded module
+            if eb_module_keys:
```

Comment:
The reason for checking for orphaned `$EBROOT*` env vars is because that's we rely on for checking for loaded (EasyBuild-generated) modules, so I prefer keeping this together in the same method.

If I flesh it out, we would need to call boths methods in sequence anyway, and we'd have to duplicate some of the code, i.e. collecting defined `$EBROOT*` env vars, and tying the knot with the loaded modules...